### PR TITLE
BFD-2447: Remove pipeline skip record functionality

### PR DESCRIPTION
**Pull request by mackec1**
**JIRA Ticket:**
[BFD-2447](https://jira.cms.gov/browse/BFD-2447)

**User Story or Bug Summary:**
Remove functionality related to the temp fix for handling CCW RIF data that co-mingled different REF_YR values; this caused the BFD pipeline to overwrite BENEFICIARY_MONTHLY data already stored. The _fix_ was to skip records whose REF_YR did not match BFD's current year (i.e., RIF record had 2022, while BFD was processing as 2023).

CCW has changed their processing and now **ONLY provides a single REF_YR** value for all Beneficiary RIF records. BFD has not encountered any RIF data with bifurcated (invalid) REF_YR values since CCW has implmeneted their fix.

### What Does This PR Do?
This PR removes support (code) for bifurcated REF_YR values in a single Beneficiary RIF file; changes include:
- Removed terraform code that created an ENV variable denoting BFD's notion of the current REF_YR; records not matching that value would be skipped.
- Removed pipeline code that checked each RIF record, allowing only those records that matched BFD's REF_YR to be stored in the db (BENEFICIARIES, BENEFICARY_MONTHLY); non-matching records were logged and stored in the SKIPPED_RIF_RECORDS table).
- removed ephemeral.yaml from the list of _sensitive files_ that are subject to the pre-commit verification to ensure no cleartext secrets are being committed to the repo.
- removed tests from RifLoaderIT that handled skipped records; renamed some tests to remove reference to skippedRecords.
- BFD (now) officially relies on CCW as the _source of truth_ for REF_YR processing; CCW has committed to only having one REF_YR value for all Benficiary RIF data.

_NOTE_: This PR does not include a flyway script to remove the SKIPPED_RIF_RECORDS table; this data may be removed at a future date depending on analysis.

### What Should Reviewers Watch For?

* Does the pipeline code and infrastructure changes meet the AC for the ticket?
* Are there any unhandled and/or untested edge cases you can think of?
* Can you find any bugs if you run the code locally and test it manually?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java [hunk 1]
```diff
@@ -122,17 +122,6 @@ public final class AppConfiguration extends BaseAppConfiguration {
...
-   * The name of the environment variable that should be used to provide the {@link
-   * LoadAppOptions#isFilteringNonNullAndNon2023Benes()} value, which is a bit complex; please see
-   * its description for details.
-   *
-   * <p>Note: This filtering option (and implementation) is an inelegant workaround, which should be
-   * removed as soon as is reasonable.
-   */
-  public static final String ENV_VAR_KEY_RIF_FILTERING_NON_NULL_AND_NON_2023_BENES =
-      "FILTERING_NON_NULL_AND_NON_2023_BENES";
-
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java:135]: Removing possibly the longest constant name ever!

File: ops/terraform/services/base/values/ephemeral.yaml [hunk 2]
```diff
@@ -1,4 +1,5 @@
 ---
+# CIPHER FILE HEADER, DO NOT REMOVE (noop preamble to satisfy the pre-commit script)
```
Comment (mjburling) [inline ops/terraform/services/base/values/ephemeral.yaml:2]: As discussed, I think we ought to remove both this line and the reference to `ops/terraform/services/base/values/ephemeral.yaml` from `.github/resources/.sensitive-values`

Re-iterating, with the usage of matching, environment-specific KMS keys with both in the repository via `cipher` and in the remote storage via AWS SSM, ephemeral-specific encrypted values aren't possible with the current implementation. I don't think we've identified a good use-case to support them yet, either.

File: apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java [hunk 3]
```diff
@@ -951,7 +931,7 @@ public void loadNonBeneficiaryWhenInsertAndFilterOnExpectRecordLoaded() {
             StaticRifResourceGroup.SAMPLE_A.getResources());
     loadSample(
         "non-Bene sample",
-        CcwRifLoadTestUtils.getLoadOptions(USE_INSERT_IDEMPOTENT_STRATEGY, FILTER_MODE_DISABLED),
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java:954]: FILTER_MODE_DISABLED should be removed a few lines above this.

File: apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java
Comment (mjburling) [inline apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java:1]: Many of these tests are specific to the skip record functionality, aren't they? Are they accurate and continue to provide value?

File: .github/resources/.sensitive-files [hunk 4]
```diff
@@ -2,7 +2,6 @@ ops/ansible/playbooks-ccs/files/client_data_server_local_test_env_test_keypair.p
 ops/ansible/playbooks-ccs/files/client_data_server_local_test_env_prod_keypair.pem
 ops/ansible/playbooks-ccs/files/client_data_server_local_test_env_dpr_keypair.pem
 ops/terraform/env/mgmt/base_config/values/mgmt.yaml
-ops/terraform/services/base/values/ephemeral.yaml
```
Comment (mjburling) [inline .github/resources/.sensitive-files:5]: 👍

**Review by brianburton — COMMENTED** [review]
Looks great.  I'll approve once the one local variable has been removed.

File: apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java [hunk 5]
```diff
@@ -1247,13 +1226,6 @@ private void validateBeneficiaryAndSkippedCountsInDatabase(
       beneCountQuery.select(criteriaBuilder.count(beneCountQuery.from(Beneficiary.class)));
       Long beneCount = entityManager.createQuery(beneCountQuery).getSingleResult();
       assertEquals(expectedBeneCount, beneCount, "Unexpected number of beneficiary records.");
-
-      // Count and verify the number of bene records in the DB.
-      CriteriaQuery<Long> skippedCountQuery = criteriaBuilder.createQuery(Long.class);
-      skippedCountQuery.select(
-          criteriaBuilder.count(skippedCountQuery.from(SkippedRifRecord.class)));
-      Long skippedCount = entityManager.createQuery(skippedCountQuery).getSingleResult();
-      assertEquals(expectedSkippedCount, skippedCount, "Unexpected number of skipped records.");
```
Comment (brianburton) [inline apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java:1256]: This makes `expectedSkippedCount` obsolete and possibly some tests that call this method as well.

File: apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java
Comment (mackec1) [reply apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java:1]: not really relevant anymore; they succeeded because 0 (zero) being passed into a generic validation count. Renamed some tests, removed those that were essentially only testing REF_YR and skipped records. Fixed the validator count to only care about beneficiaries records.

File: apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java (hunk 3 above)
Comment (mackec1) [reply apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java:954]: done (along with renaming and/or removing tests particular to REF_YR counts and skipped records.

File: apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java (hunk 5 above)
Comment (mackec1) [reply apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/load/RifLoaderIT.java:1256]: yup...rename tests and removed others.

**Review by brianburton — APPROVED** [review]

**Review by brandoncruz3 — APPROVED** [review]
